homeassistant/components/openalpr.py

```diff
@@ -0,0 +1,488 @@
+                interval=device.get(CONF_INTERVAL),
+                api=alpr_api,
+                input_source=input_source,
+                username=device.get(CONF_USERNAME),
+                password=device.get(CONF_PASSWORD),
+            )
+
+        # register shutdown event
+        openalpr_device.append(alpr_dev)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, alpr_dev.shutdown)
+
+    component.add_entities(openalpr_device)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    def _handle_service_scan(service):
+        """Handle service for immediately scan."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.scan()
+
+    hass.services.register(DOMAIN, SERVICE_SCAN,
+                           _handle_service_scan,
+                           descriptions[DOMAIN][SERVICE_SCAN],
+                           schema=SERVICE_SCAN_SCHEMA)
+
+    # Add restart service only if a device use ffmpeg as render
+    if not use_render_fffmpeg:
+        return True
+
+    def _handle_service_restart(service):
+        """Handle service for restart ffmpeg process."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.restart()
+
+    hass.services.register(DOMAIN, SERVICE_RESTART,
+                           _handle_service_restart,
+                           descriptions[DOMAIN][SERVICE_RESTART],
+                           schema=SERVICE_RESTART_SCHEMA)
+
+    return True
+
+
+class OpenalprDevice(Entity):
+    """Represent a openalpr device object for processing stream/images."""
+
+    def __init__(self, name, interval, api):
+        """Init image processing."""
+        self._name = name
+        self._interval = interval
+        self._api = api
+        self._last = {}
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        confidence = 0
+        plate = STATE_UNKNOWN
+
+        # search high plate
+        for i_pl, i_co in self._last.items():
+            if i_co > confidence:
+                confidence = i_co
+                plate = i_pl
+        return plate
+
+    def shutdown(self, event):
+        """Close stream."""
+        if hasattr(self._api, "shutdown"):
+            self._api.shutdown(event)
+
+    def restart(self):
+        """Restart stream."""
+        pass
+
+    def _process_image(self, image):
+        """Callback for processing image."""
+        self._api.process_image(image, self._process_event)
+
+    def _process_event(self, plates):
+        """Send event with new plates."""
+        state_change = False
+        plates_set = set(plates)
+        last_set = set(self._last)
+        new_plates = plates_set - last_set
+
+        # send events
+        for i_plate in new_plates:
+            self.hass.bus.fire(EVENT_FOUND, {
```

Comment:
I don't see any benefit for trigger Automation if we have confidence. But for the future, maybe if the user want that.